src/aof.c

```diff
@@ -671,7 +671,18 @@ struct client *createAOFClient(void) {
     c->argv = NULL;
     c->argv_len_sum = 0;
     c->bufpos = 0;
-    c->flags = 0;
+
+    /*
+     * The AOF client should never be blocked (unlikle master
+     * replication connection).
+     * This is because blocking the AOF client might cause
+     * deadlock (because potentially no one will unblock it).
+     * Also, if the AOF client will be blocked just for
+     * background processing there is a chance that the
+     * command execution order will be violated.
+     */
+    c->flags = CLIENT_DENY_BLOCKING;
```

Comment:
@MeirShpilraien thinking of it again, this flag can cause a command to return an error or exit with a timeout response right?

so this will silence the assertion you showed me earlier:
```
serverAssert((fakeClient->flags & CLIENT_BLOCKED) == 0);
```
maybe it's better to avoid setting this flag (and keep a comment explaining it)?

Reply:
@oranagra if we will not set it then modules will have to check for both DENY_BLOCKING and LOADING flags instead of just the DENY_BLOCKING. Also, this assert will catch modules (and commands) that ignore the DENY_BLOCKING flag. I think that if we want to catch commands that are not allowed inside AOF loading, we should do it with a new DISALLOW_ON_AOF flag (because there are more commands that should not reach the AOF and those are not necessarily blocking commands), what do you think?

Reply:
i suppose that's ok to leave as is then.
in any case, if we want to check that AOF didn't fail executing the command we may need to check their response. i'm guessing that assertion wasn't really needed.
and you're right that it still served (some) purpose.

Reply:
Regardless I now realize, we should put such assert right after executing the command on RM_Call/MULTI/LUA, what do you think?

Reply:
sounds like a good idea.